client/app/telecom/telephony/service/consumption/incomingCalls/telecom-telephony-service-consumption-incomingCalls.controller.js

```diff
@@ -1,34 +1,21 @@
-angular.module('managerApp').controller('TelecomTelephonyServiceConsumptionIncomingCallsCtrl', function ($stateParams, $q, $translate, $filter, $timeout, OvhApiTelephony, ToastError) {
+angular.module('managerApp').controller('TelecomTelephonyServiceConsumptionIncomingCallsCtrl', function ($stateParams, $q, $translate, $filter, $timeout, OvhApiTelephony, ToastError, voipService) {
   const self = this;
 
   function fetchIncomingConsumption() {
-    return OvhApiTelephony.Service().VoiceConsumption().v6().query({
+    return voipService.getServiceConsumption({
       billingAccount: $stateParams.billingAccount,
       serviceName: $stateParams.serviceName,
-    }).$promise.then(ids => $q
-      .all(_.map(
-        _.chunk(ids, 50),
-        chunkIds => OvhApiTelephony.Service().VoiceConsumption().v6().getBatch({
-          billingAccount: $stateParams.billingAccount,
-          serviceName: $stateParams.serviceName,
-          consumptionId: chunkIds,
-        }).$promise,
-      ))
-      .then(chunkResult => _.flatten(chunkResult)))
-      .then(result => _.chain(result)
-        .pluck('value')
-        .filter(conso => conso.wayType !== 'outgoing')
-        .map((conso) => {
-          _.set(conso, 'durationAsDate', new Date(conso.duration * 1000));
-          if (conso.wayType === 'incoming' && conso.duration === 0) {
-            _.set(conso, 'wayType', 'missing');
-          }
-          if (/anonymous/.test(conso.calling)) {
-            _.set(conso, 'calling', $translate.instant('telephony_service_consumption_anonymous'));
-          }
-          return conso;
-        })
-        .value());
+    }).then(result => result.filter(conso => conso.wayType !== 'outgoing')
```

Comment:
`filter(conso => conso.wayType !== 'outgoing').map((conso) => _.set(conso, 'durationAsDate', new Date(conso.duration * 1000)))` could be in a service has it come at least 3 times

Reply:
I've just limited refactor to common code between telecom-telephony-service-consumption-incomingCalls.controller.js & telecom-telephony-service-consumption-outgoingCalls.controller.js

These two files will be refactor later

Reply:
Alright, next time don't hesitate to comment your PR so we have all the data beforehand 😉

Reply:
Yes I'll write it directly on top of the concerned files :)